### ISSUE TITLE
Changes to behaviour of Item

### ISSUE THREAD

**Issue by gilesknap** [conqp/mcipc#41]
In the latest version of mcipc there have been some changes to behaviour of Item.

I could fix my code to work with the new behaviour  but I'd like to understand if these are intentional. 

```python
# pip install mcipc==2.3.8 rcon==2.2.1

In [2]: str(Item.AIR)
Out[2]: 'minecraft:air'

In [3]: Item("minecraft:air")
ValueError: 'minecraft:air' is not a valid Item

In [4]: Item("air")
Out[4]: <Item.AIR: 'air'>

#pip install mcipc==2.4.0 rcon==2.3.1

In [2]: str(Item.AIR)
Out[2]: 'Item.AIR'

In [3]: Item("minecraft:air")
Out[3]: <Item.AIR: 'minecraft:air'>

In [4]: Item("air")
ValueError: 'air' is not a valid Item
```

**Comment by gilesknap** [comment]
My preferred behaviour would be that Item __str__ returns the short form but the constructor could take the long or short form:
```
In [2]: str(Item.AIR)
Out[2]: 'air'

In [4]: Item("air")
Out[4]: <Item.AIR: 'minecraft:air'>

In [4]: Item("minecraft:air")
Out[4]: <Item.AIR: 'minecraft:air'>
```

**Comment by conqp** [comment]
Yep. Unintentionally my last change to this Enum caused an unintended change of the previous behavior.
I reverted the changes and also added regression tests.
Fixed in 19a22f907841ae8b1de69a289819c35f880d83d7

**Comment by gilesknap** [comment]
@conqp for some reason this is still breaking the mcwb modules tests.

There is a change in behaviour which is affecting the vaildate function https://github.com/mcipc-tools/mcwb/blob/master/mcwb/functions.py#L88-L101.

When casting an array of Item to a numpy array the numpy array now ends up containing str instead of Item and this breaks the validation. Its not clear to me why this is happening.

Investigating further.

**Comment by gilesknap** [comment]
downgrading mcipc and rcon does fix the issue.

**Comment by conqp** [comment]
>

**Comment by gilesknap** [comment]
OK the subclass of ``str`` is probably what is doing it. numpy likes to use the base class of its elements as its type, I believe.
It also means that ndarray.toList() is returning a list of str even if you do make a numpy array of Item.

I have worked around this by supplying a dtype of Item to numpy constructors, so mcwb is working again. Now when we save an 'Items' object to a json file it now looks like a list of long form string (because it uses toList), but this works OK and is a bit more legible than the old format. loading of Items objects still supports the old and new json formats.

In short, I think the new behaviour is now fine for mcwb and mciwb. The only problem is if there are any other users relying on the old behaviour.
